modules/tinymce/src/core/test/ts/browser/lookup/UserLookupTest.ts

```diff
@@ -0,0 +1,157 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { Fun } from '@ephox/katamari';
+import { TinyHooks } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import type Editor from 'tinymce/core/api/Editor';
+import type { User } from 'tinymce/core/lookup/UserLookup';
+
+describe('browser.tinymce.core.UserLookupTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+    current_user_id: 'test-user-1',
+    fetch_user_by_id: (userId: string): Promise<User> => {
+      return new Promise((resolve) => {
+        setTimeout(() => {
+          resolve({
+            id: userId,
+            name: 'Test User',
+            avatar: 'test-avatar.png',
+            description: 'Test Description',
+          } as User);
+        }, 100);
+      });
+    },
+  });
+
+  it('Should return the configured user ID', () => {
+    const editor = hook.editor();
+    const currentUserId = editor.userLookup.getCurrentUserId();
+    assert.equal(currentUserId, 'test-user-1', 'Should return the configured user ID');
+  });
+
+  it('Should fetch user information', async () => {
+    const editor = hook.editor();
+    const userId = 'test-user-1';
+
+    const user = await editor.userLookup
+      .fetchUserById(userId)
+      .then(Fun.identity)
+      .catch(() => {
+        assert.fail('Should not throw an error');
+      });
+
+    assert.equal(user.id, userId, 'Should return user with correct ID');
+    assert.equal(user.name, 'Test User', 'Should return user with correct name');
+    assert.equal(user.avatar, 'test-avatar.png', 'Should return user with correct avatar');
+    assert.equal(user.description, 'Test Description', 'Should return user with correct description');
+  });
+
+  it('Should return cached user data on second call', async () => {
+    const editor = hook.editor();
+    const userId = 'test-user-1';
+
+    // Fetch and cache
+    const firstCall = await editor.userLookup.fetchUserById(userId);
+
+    // Return cached data
+    const secondCall = await editor.userLookup.fetchUserById(userId);
+
+    assert.deepEqual(firstCall, secondCall, 'Should return same user data from cache');
+  });
+
+  it('Should handle errors gracefully', async () => {
+    const editor = hook.editor();
+
+    // Override fetch_user_by_id to simulate an error
+    editor.options.set('fetch_user_by_id', () => Promise.reject('User not found'));
```

Comment:
I wonder if we should `normalize` the string to an error type since promises are weird so maybe we could enforce them to not be as weird by type guards here. Since we we add validations on resolved values we could add normalization on rejected values. Something like `(err) => err instanceof Error ? err : new Error(String(err))` or something.

Reply:
It probably depends what we plan to do with the rejection. But the general advice seems to be yes we should do that.
https://stackoverflow.com/a/26020867/7546